### PR TITLE
github: Added CI for linting and compiling checks

### PR DESCRIPTION
**Pull request by Nykseli**
## Detailed description

Adding automated checks for lint rules and making sure that the project compiles reduces the chance of incorrect code finding its way to the code base.

## Your checklist for this pull request

* [x] I've read the [Code of Conduct](https://github.com/mangrove-lang/mangrove-vscode/blob/main/CODE_OF_CONDUCT.md)
* [x] I've read the [guidelines for contributing](https://github.com/mangrove-lang/mangrove-vscode/blob/main/CONTRIBUTING.md) to this repository
* [x] It builds and lints cleanly
* [x] I've tested it to the best of my ability
* [x] My commit messages provide a useful short description of what the commits do
* [x] I've made sure that my commits are [signed/verified](https://docs.github.com/en/authentication/managing-commit-signature-verification/about-commit-signature-verification)

### CODE REVIEW

File: .github/workflows/checks.yml [hunk 1]
```diff
@@ -0,0 +1,63 @@
+name: Checks
```
Comment (dragonmux) [inline .github/workflows/checks.yml:1]: nitpick: Please use a name such as "PR checks" so we know at a glance that this is specific the PR flow

File: .github/workflows/checks.yml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    branches:
+      - 'main'
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.job }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  check-linting:
```
Comment (dragonmux) [inline .github/workflows/checks.yml:17]: Given how GH builds the display string, this'll wind up being "Checks / check-linting" - might as well drop the `check-` prefix? ie "Checks / linting" or just "Checks / lint"

This is up to you though

File: .github/workflows/checks.yml [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        with:
+          node-version: ${{ matrix.node-version }}
+
+      - name: Install dependencies
+        run: npm ci
+
+      - name: Run eslint
+        run: npm run lint
+
+  check-compiling:
```
Comment (dragonmux) [inline .github/workflows/checks.yml:41]: Same nitpick as above

**Review by dragonmux — CHANGES_REQUESTED** [review]
Technically this PR looks perfect, thank you for the contribution!

The nitpicks below are all linguistic and user-facing terminology in nature.

The only additional one we couldn't turn into a review note is that rather than "checks.yml", we would rather you ran with "pr-checks.yml" or "pr-workflow.yml" or similar so it's more obvious from outside the file that the flow is specifically for PRs to give us naming space for additional workflows that'll run on main

**Comment by Nykseli** [comment]
All review items should now be addressed

**Review by dragonmux — APPROVED** [review]
LGTM, merging

**Pull request merged**